### PR TITLE
deactivate a site

### PR DESCRIPTION
**Pull request by amirbey**
- rename site.buildstatus to site.isactive
- prevent users from updating/rebuilding a site
- display the site is inactive in ui
- allow updating site status in the admin ui

resolves https://github.com/18F/federalist/issues/3671

### CODE REVIEW

**Comment by davemcorwin** [comment]
@amirbey what is the advantage to renaming the `buildState` column and changing it to a boolean?

**Comment by amirbey** [comment]
> @amirbey what is the advantage to renaming the `buildState` column and changing it to a boolean?

 the current prod implementation only ignores webhooks ... the intent is to not only not build the site from push webhooks  but also make the site inaccessible in the app same as organizations that are deactivated.
the field wasn't being used at all so there was not real risk in changing it ... seems more accurate

**Comment by davemcorwin** [comment]
> the current prod implementation only ignores webhooks ... the intent is to not only not build the site from push webhooks but also make the site inaccessible in the app same as organizations that are deactivated. 

renaming/changing to boolean doesn't affect our ability to do this.

> the field wasn't being used at all so there was not real risk in changing it ... seems more accurate

yes, but how does this help?

**Comment by amirbey** [comment]
> > the current prod implementation only ignores webhooks ... the intent is to not only not build the site from push webhooks but also make the site inaccessible in the app same as organizations that are deactivated.
> 
> renaming/changing to boolean doesn't affect our ability to do this.
> 
> > the field wasn't being used at all so there was not real risk in changing it ... seems more accurate
> 
> yes, but how does this help?

i agree ... it does not affect ability to do this .. 👍🏿 
while going forward,  the field would be doing more than deactivating builds and mimicing the behavior if organizations.isActive, is there detriment to renaming it to to something not limited to builds?

File: api/services/Webhooks.js [hunk 1]
```diff
@@ -17,7 +17,7 @@ const findSiteForWebhookRequest = (payload) => {
   });
 };
 
-const shouldBuildForSite = site => site && site.buildStatus !== 'inactive'
+const shouldBuildForSite = site => site && site.isActive
```
Comment (davemcorwin) [inline api/services/Webhooks.js:20]:
```suggestion
const shouldBuildForSite = site => site?.isActive
```

**Review by davemcorwin — APPROVED** [review]
One fancy js suggestion below, otherwise 💯

**Pull request merged**